cpp/ScanChain_Zp.C

```diff
@@ -591,21 +748,24 @@ int ScanChain(TChain *ch, double genEventSumw, TString year, TString process) {
       vector<int> extra_electrons;
       for ( int i = 0; i < nt.nElectron(); i++ ){
         if ( nt.Electron_pt().at(i) > 10. &&
-            fabs(nt.Electron_eta().at(i)) < 2.4 &&
-            nt.Electron_cutBased().at(i) > 0 && 
-            nt.Electron_miniPFRelIso_all().at(i) < 0.1){
+	     fabs(nt.Electron_eta().at(i)) < 2.5 &&
+	     nt.Electron_cutBased().at(i) > 0 && 
+	     nt.Electron_miniPFRelIso_all().at(i) < 0.1 &&
+	     fabs(nt.Electron_dxy().at(i)) < 0.2 &&
+	     fabs(nt.Electron_dz().at(i)) < 0.5 ) {
           extra_electrons.push_back(i);
         }
       }
 
       // IsoTracks
       vector<int> extra_isotracks_lep;
       for ( int i = 0; i < nt.nIsoTrack(); i++ ){
-        if ( nt.IsoTrack_isPFcand().at(i) && 
-            nt.IsoTrack_pt().at(i) > 5. && (abs(nt.IsoTrack_pdgId().at(i))==11 || (abs(nt.IsoTrack_pdgId().at(i))==13)) &&
-            fabs(nt.IsoTrack_eta().at(i)) < 2.4 &&
-            fabs(nt.IsoTrack_dz().at(i)) < 0.1 &&
-            nt.IsoTrack_pfRelIso03_chg().at(i) < 0.2){
+        if ( nt.IsoTrack_isPFcand().at(i) &&
+	     (abs(nt.IsoTrack_pdgId().at(i))==11 || (abs(nt.IsoTrack_pdgId().at(i))==13)) &&
+	     nt.IsoTrack_pt().at(i) > 5. && 
+	     fabs(nt.IsoTrack_eta().at(i)) < 2.5 &&
+	     fabs(nt.IsoTrack_dz().at(i)) < 0.1 &&
```

Comment:
Just noticed: For consistency, should we add a d_xy cut here, like we do for extra muons and electrons? Also in the isotrack charged hadrons?

Reply:
Adding it.

Reply:
Added.